packages/material-ui-styles/src/ThemeProvider/ThemeProvider.js

```diff
@@ -24,7 +51,7 @@ function mergeOuterLocalTheme(outerTheme, localTheme) {
     return mergedTheme;
   }
 
-  return { ...outerTheme, ...localTheme };
+  return mergeThemes(outerTheme, localTheme);
```

Comment:
The scenario that I was testing was, having two `ThemeProvider`s, each of them adding new variants, but the new theme (by accident I guess :)) did not have the prev theme in it's parameter, so even with this utility it didn't work. Later I noticed that by this spread, the new will always override the previous...

Reply:
I propose we resolve this problem in another thread. I believe it's not directly linked to what we are going after. I would also make the case it can also be considered an issue today. Well spotted.